crates/router/src/connector/braintree/transformers.rs

```diff
@@ -254,234 +549,1255 @@ impl From<BraintreePaymentStatus> for enums::AttemptStatus {
+                        item.data.get_payment_method_token()?,
+                        item.data.request.payment_method_data.clone(),
+                    )?),
+                    mandate_reference: None,
+                    connector_metadata: None,
+                    network_txn_id: None,
+                    connector_response_reference_id: None,
+                    incremental_authorization_allowed: None,
+                    charge_id: None,
+                }),
+                ..item.data
+            }),
+        }
     }
 }
 
-impl<F, T>
+impl<F>
     TryFrom<
-        types::ResponseRouterData<F, BraintreeSessionTokenResponse, T, types::PaymentsResponseData>,
-    > for types::RouterData<F, T, types::PaymentsResponseData>
+        types::ResponseRouterData<
+            F,
+            BraintreeCompleteChargeResponse,
+            types::CompleteAuthorizeData,
+            types::PaymentsResponseData,
+        >,
+    > for types::RouterData<F, types::CompleteAuthorizeData, types::PaymentsResponseData>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
         item: types::ResponseRouterData<
             F,
-            BraintreeSessionTokenResponse,
-            T,
+            BraintreeCompleteChargeResponse,
+            types::CompleteAuthorizeData,
             types::PaymentsResponseData,
         >,
     ) -> Result<Self, Self::Error> {
-        Ok(Self {
-            response: Ok(types::PaymentsResponseData::SessionResponse {
-                session_token: api::SessionToken::Paypal(Box::new(
-                    payments::PaypalSessionTokenResponse {
-                        session_token: item.response.client_token.value.expose(),
-                        connector: "braintree".to_string(),
-                        sdk_next_action: payments::SdkNextAction {
-                            next_action: payments::NextActionCall::Confirm,
-                        },
-                    },
-                )),
+        match item.response {
+            BraintreeCompleteChargeResponse::ErrorResponse(error_response) => Ok(Self {
+                response: build_error_response(&error_response.errors.clone(), item.http_code),
+                ..item.data
             }),
-            ..item.data
-        })
+            BraintreeCompleteChargeResponse::PaymentsResponse(payment_response) => {
+                let transaction_data = payment_response.data.charge_credit_card.transaction;
+                let status = enums::AttemptStatus::from(transaction_data.status.clone());
+                let response = if let enums::AttemptStatus::Failure = status {
+                    Err(types::ErrorResponse {
+                        code: transaction_data.status.to_string().clone(),
+                        message: transaction_data.status.to_string().clone(),
+                        reason: None,
+                        attempt_status: None,
+                        connector_transaction_id: None,
```

Comment:
Can we map `transaction_data.id` to `connector_transaction_id` here ?

Reply:
mapped